torch/_torch_docs.py

```diff
@@ -7951,10 +7951,17 @@ def merge_dicts(*dicts):
 dimension :attr:`dim` is input from :attr:`start` to ``start + length``. The
 returned tensor and :attr:`input` tensor share the same underlying storage.
 
+:attr:`start` accepts a `Tensor` for models that have it as such before calling
+`narrow`.  This avoids the call to `item()` for backends where it is expensive
+(like XLA).
+
 Args:
     input (Tensor): the tensor to narrow
-    dim (int): the dimension along which to narrow
-    start (Tensor or int): the starting dimension
+    dim (int): the dimension along which to narrow. Can be negative, which means
+        counting from the right
+    start (Tensor or int): the starting dimension. If `Tensor`, it must be an
+        0-dim integral `Tensor` (bools not allowed). Can be negative, which
```

Comment:
"counting from the right" isn't clear enough here. Let's be more precise about what a negative start value means

Reply:
Also: "the starting dimension"? Doesn't `dim` specify the dimension? I thought `start` and `length` specify the part of that dimension to narrow to?

Reply:
Updated, LMK if the wording can be improved further.